src/inventorymanager.cpp

```diff
@@ -842,7 +842,8 @@ void ICraftAction::apply(InventoryManager *mgr,
 			count_remaining--;
 
 		// Get next crafting result
-		found = getCraftingResult(inv_craft, crafted, temp, false, gamedef);
+		if (!getCraftingResult(inv_craft, crafted, temp, false, gamedef))
+			break;
```

Comment:
Yes, but found is overwritten two lines later.  It could have been an oversight in the original code, and because it worked when tested, it was never fixed.